engine/src/main/java/org/hibernate/search/query/dsl/impl/AbstractFacet.java

```diff
@@ -39,9 +41,21 @@ public String getValue() {
 		return value;
 	}
 
+	/**
+	 * @deprecated Use {@link #getFacetFieldName()} instead.
+	 */
 	@Override
+	@Deprecated
 	public String getFieldName() {
-		return fieldName;
+		return getFacetFieldName();
+	}
+
+	public String getFacetFieldName() {
+		return facetFieldName;
+	}
+
+	public String getSourceFieldName() {
```

Comment:
In this specific case, we are actually speaking about a field... See [here](https://github.com/hibernate/hibernate-search/pull/1360/files/8a1cb838838a7435b2e2787e5766ea2d788884bb#diff-2bb3b66e5fbcd04002af53e6f647d57bR33) and the `getFacetQuery` method a few lines below to see why.

Maybe "Source" isn't the right term, though. "Attached"? "Origin"? I'm not sure how to call it. It's basically whatever we pass to `@Facet.forField`...